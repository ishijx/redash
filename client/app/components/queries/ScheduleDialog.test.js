import React from 'react';
import { mount } from 'enzyme';
import { ScheduleDialog } from './ScheduleDialog';

const defaultProps = {
  show: true,
  query: {
    schedule: {
      time: null,
      until: null,
      interval: null,
      day_of_week: null,
    },
  },
  refreshOptions: [
    60, 300, 600, // 1, 5 ,10 mins
    3600, 36000, 82800, // 1, 10, 23 hours
    86400, 172800, 518400, // 1, 2, 6 days
    604800, 1209600, // 1, 2, 4 weeks
  ],
  updateQuery: () => {},
  onClose: () => {},
};

function getWrapper(schedule = {}, props = {}) {
  const defaultSchedule = defaultProps.query.schedule;
  props = Object.assign(
    {},
    defaultProps,
    props,
    { query: { schedule: Object.assign({}, defaultSchedule, schedule) } },
  );
  return [mount(<ScheduleDialog {...props} />), props];
}

describe('ScheduleDialog', () => {
  describe('Sets correct schedule settings', () => {
    test('Sets to "Never"', () => {
      const [wrapper] = getWrapper();
      const el = wrapper.find('[data-testid="interval"]');
      expect(el).toMatchSnapshot();
    });
  });
});
