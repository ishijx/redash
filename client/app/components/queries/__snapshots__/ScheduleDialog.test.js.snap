// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleDialog Sets correct schedule settings Sets to "Never" 1`] = `
<div
  data-testid="interval"
>
  <Select
    choiceTransitionName="zoom"
    className="input"
    dropdownMatchSelectWidth={false}
    getPopupContainer={[Function]}
    onChange={[Function]}
    prefixCls="ant-select"
    showSearch={false}
    transitionName="slide-up"
    value="Never"
  >
    <LocaleReceiver
      componentName="Select"
      defaultLocale={
        Object {
          "notFoundContent": "Not Found",
        }
      }
    >
      <Select
        allowClear={false}
        autoClearSearchValue={true}
        backfill={false}
        choiceTransitionName="zoom"
        className="input"
        combobox={false}
        defaultActiveFirstOption={true}
        defaultOpen={false}
        dropdownMatchSelectWidth={false}
        dropdownMenuStyle={Object {}}
        dropdownStyle={Object {}}
        getPopupContainer={[Function]}
        labelInValue={false}
        multiple={false}
        notFoundContent="Not Found"
        onBlur={[Function]}
        onChange={[Function]}
        onDeselect={[Function]}
        onFocus={[Function]}
        onInputKeyDown={[Function]}
        onSearch={[Function]}
        onSelect={[Function]}
        optionFilterProp="value"
        optionLabelProp="children"
        placeholder=""
        prefixCls="ant-select"
        showAction={
          Array [
            "click",
          ]
        }
        showArrow={true}
        showSearch={false}
        tags={false}
        tokenSeparators={Array []}
        transitionName="slide-up"
        value="Never"
      >
        <SelectTrigger
          combobox={false}
          defaultActiveFirstOption={true}
          dropdownMatchSelectWidth={false}
          dropdownMenuStyle={Object {}}
          dropdownStyle={Object {}}
          getPopupContainer={[Function]}
          inputValue=""
          multiple={false}
          onDropdownVisibleChange={[Function]}
          onMenuDeselect={[Function]}
          onMenuSelect={[Function]}
          onPopupFocus={[Function]}
          options={Array []}
          prefixCls="ant-select"
          showAction={
            Array [
              "click",
            ]
          }
          showSearch={false}
          transitionName="slide-up"
          value={
            Array [
              "Never",
            ]
          }
          visible={false}
        >
          <Trigger
            action={Array []}
            afterPopupVisibleChange={[Function]}
            blurDelay={0.15}
            builtinPlacements={
              Object {
                "bottomLeft": Object {
                  "offset": Array [
                    0,
                    4,
                  ],
                  "overflow": Object {
                    "adjustX": 0,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tl",
                    "bl",
                  ],
                },
                "topLeft": Object {
                  "offset": Array [
                    0,
                    -4,
                  ],
                  "overflow": Object {
                    "adjustX": 0,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "bl",
                    "tl",
                  ],
                },
              }
            }
            combobox={false}
            defaultActiveFirstOption={true}
            defaultPopupVisible={false}
            destroyPopupOnHide={false}
            dropdownMatchSelectWidth={false}
            dropdownMenuStyle={Object {}}
            dropdownStyle={Object {}}
            focusDelay={0}
            getDocument={[Function]}
            getPopupClassNameFromAlign={[Function]}
            getPopupContainer={[Function]}
            hideAction={
              Array [
                "click",
              ]
            }
            inputValue=""
            mask={false}
            maskClosable={true}
            mouseEnterDelay={0}
            mouseLeaveDelay={0.1}
            multiple={false}
            onDropdownVisibleChange={[Function]}
            onMenuDeselect={[Function]}
            onMenuSelect={[Function]}
            onPopupAlign={[Function]}
            onPopupVisibleChange={[Function]}
            options={Array []}
            popup={
              <DropdownMenu
                defaultActiveFirstOption={true}
                dropdownMenuStyle={Object {}}
                inputValue=""
                menuItems={Array []}
                multiple={false}
                onMenuDeselect={[Function]}
                onMenuSelect={[Function]}
                onPopupFocus={[Function]}
                prefixCls="ant-select-dropdown"
                value={
                  Array [
                    "Never",
                  ]
                }
                visible={false}
              />
            }
            popupAlign={Object {}}
            popupClassName="ant-select-dropdown--single"
            popupPlacement="bottomLeft"
            popupStyle={Object {}}
            popupTransitionName="slide-up"
            popupVisible={false}
            prefixCls="ant-select-dropdown"
            showAction={
              Array [
                "click",
              ]
            }
            showSearch={false}
            transitionName="slide-up"
            value={
              Array [
                "Never",
              ]
            }
            visible={false}
          >
            <div
              className="input ant-select ant-select-enabled"
              key="trigger"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <div
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup="true"
                className="ant-select-selection
            ant-select-selection--single"
                key="selection"
                onKeyDown={[Function]}
                role="combobox"
                tabIndex={0}
              >
                <div
                  className="ant-select-selection__rendered"
                >
                  <div
                    className="ant-select-selection-selected-value"
                    key="value"
                    style={
                      Object {
                        "display": "block",
                        "opacity": 1,
                      }
                    }
                    title="Never"
                  >
                    Never
                  </div>
                </div>
                <span
                  className="ant-select-arrow"
                  key="arrow"
                  onClick={[Function]}
                  style={
                    Object {
                      "WebkitUserSelect": "none",
                      "userSelect": "none",
                    }
                  }
                  unselectable="on"
                >
                  <b />
                </span>
              </div>
            </div>
          </Trigger>
        </SelectTrigger>
      </Select>
    </LocaleReceiver>
  </Select>
</div>
`;
